src/hls-adapter.js

```diff
@@ -668,9 +676,11 @@ export default class HlsAdapter extends BaseMediaSourceAdapter {
       this._hls.startLoad(this._startTime);
     }
     this._playerTracks = this._parseTracks();
-    //TODO: workaround for various hls.js issues with the initial track selection logic.
-    //TODO: once https://github.com/video-dev/hls.js/issues/1948 is solved and we move to next hls.js version we need to reomve this if clause
-    if (!this._maybeHandleInitialTracksWorkaround()) {
+    if (this._hls.audioTracks.length > 1) {
```

Comment:
you can remove this audio workaround as well, need to verify that https://github.com/video-dev/hls.js/pull/1853 solves this.
you can basically only call `this._resolveLoad({tracks: this._playerTracks});`

Reply:
and delete `_maybeHandleInitialTracksWorkaround`.
And also as you noticed :-) need to take `this._maybeApplyAbrRestrictions();` call and put it back before calling resolve